
 uclibc.org master commits
 5b174cbf6597468a3e78868ff175afd6579adcd7 - use proper macro (no obj-code changes)
 435471db8561e4686e5921b7f719ab6d5a0d06f7 ldso: use ADDR_ALIGN instead of hard-coded value

---
diff --git a/ldso/ldso/dl-elf.c b/ldso/ldso/dl-elf.c
--- a/ldso/ldso/dl-elf.c
+++ b/ldso/ldso/dl-elf.c
@@ -442,8 +442,8 @@ struct elf_resolve *_dl_load_elf_shared_
 
 	DL_CHECK_LIB_TYPE (epnt, piclib, _dl_progname, libname);
 
-	maxvma = (maxvma + ADDR_ALIGN) & ~ADDR_ALIGN;
-	minvma = minvma & ~0xffffU;
+	maxvma = (maxvma + ADDR_ALIGN) & PAGE_ALIGN;
+	minvma = minvma & ~ADDR_ALIGN;
 
 	flags = MAP_PRIVATE /*| MAP_DENYWRITE */ ;
 	if (!piclib)
--
cgit v0.8.2.1
